### PR TITLE
Add HUDNotifications and Notification

### PR DESCRIPTION
**Pull request by zax-ftw**
Added `Notification` class definition. Updated `HUDNotifications` class definition. Fixed CMake generation error when running CMake from inside VS - `sourceDir/build`, otherwise it defaults to `C:<path with spaces>` and errors. Reverted `VCPKG_ROOT` name for consistency with previous versions and documentation (Example plugin, Wiki etc).

### CODE REVIEW

File: src/RE/H/HUDNotifications.cpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#include "RE/H/HUDNotifications.h"
+
+namespace RE
+{
+	HUDNotifications::Notification::Notification(const Notification& a_rhs)
```
Comment (Qudix) [inline src/RE/H/HUDNotifications.cpp:5]:
```suggestion
```
It would be preferable not to create additional .cpp files and have this in the header instead.

**Review by Qudix — CHANGES_REQUESTED** [review]

File: CMakePresets.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
       "cacheVariables": {
         "CMAKE_TOOLCHAIN_FILE": {
           "type": "STRING",
-          "value": "$env{VCPKG_INSTALLATION_ROOT}/scripts/buildsystems/vcpkg.cmake"
+          "value": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake"
```
Comment (Ryan-rsm-McKenzie) [inline CMakePresets.json:25]: Please do not change this variable, this is used for the ci

File: include/RE/H/HUDNotifications.h [hunk 3]
```diff
@@ -12,29 +15,24 @@ namespace RE
...
-				void*         unk00;  // 00
-				std::uint64_t unk08;  // 08
-				void*         unk10;  // 10
-				std::uint64_t unk18;  // 18
-			};
-			static_assert(sizeof(UnkData2) == 0x20);
+		public:
+			Notification() :
+				type(0), quest(nullptr), word(nullptr), time(0) {}
+			Notification(const Notification& a_rhs);
```
Comment (Ryan-rsm-McKenzie) [inline include/RE/H/HUDNotifications.h:23]: There's not need to define a copy ctor here, the compiler generated one is perfectly fine.

File: include/RE/H/HUDNotifications.h [hunk 4]
```diff
@@ -12,29 +15,24 @@ namespace RE
...
-			{
-				void*         unk00;  // 00
-				std::uint64_t unk08;  // 08
-				void*         unk10;  // 10
-				std::uint64_t unk18;  // 18
-			};
-			static_assert(sizeof(UnkData2) == 0x20);
+		public:
+			Notification() :
+				type(0), quest(nullptr), word(nullptr), time(0) {}
```
Comment (Ryan-rsm-McKenzie) [inline include/RE/H/HUDNotifications.h:22]: Please use default member initializers instead of defining a ctor.

**Review by Ryan-rsm-McKenzie — CHANGES_REQUESTED** [review]

File: include/RE/H/HUDNotifications.h (hunk 4 above)
Comment (zax-ftw) [reply include/RE/H/HUDNotifications.h:22]: @Ryan-rsm-McKenzie if I remember correctly, passing the `Notification` object that wasn't initialized to `0/nullptr/nullptr/0` to some related internal functions used to crash the game. Are you sure you want this change?

Comment (Ryan-rsm-McKenzie) [reply include/RE/H/HUDNotifications.h:22]: I suggest you read up on [default member initialization](https://en.cppreference.com/w/cpp/language/data_members#Member_initialization)

File: include/RE/H/HUDNotifications.h [hunk 5]
```diff
@@ -12,29 +15,19 @@ namespace RE
...
-			std::uint64_t      unk58;  // 58
+			BSString                text;             // 00 - size 10
+			BSString                status;           // 10 - size 10
+			BSFixedString           sound;            // 20 - size 08
+			BSTArray<BSFixedString> objectives;       // 28 - size 18
+			std::uint32_t           type = 0;         // 40
+			std::uint8_t            pad[4];           // 44
+			TESQuest*               quest = nullptr;  // 48
+			TESWordOfPower*         word = nullptr;   // 50
+			std::uint32_t           time = 0;         // 58 - gameTime + iObjectivesWaitTime
```
Comment (Ryan-rsm-McKenzie) [inline include/RE/H/HUDNotifications.h:28]:
```suggestion
			std::uint32_t           type{ 0 };         // 40
			std::uint32_t            pad44{ 0 };           // 44
			TESQuest*               quest{ nullptr };  // 48
			TESWordOfPower*         word{ nullptr };   // 50
			std::uint32_t           time{ 0 };         // 58 - gameTime + iObjectivesWaitTime
```

File: include/RE/H/HUDNotifications.h [hunk 6]
```diff
@@ -12,29 +15,19 @@ namespace RE
...
-			BSFixedString      unk20;  // 20
-			BSTArray<UnkData2> unk28;  // 28
-			std::uint64_t      unk40;  // 40
-			std::uint64_t      unk48;  // 48
-			std::uint64_t      unk50;  // 50
-			std::uint64_t      unk58;  // 58
+			BSString                text;             // 00 - size 10
+			BSString                status;           // 10 - size 10
+			BSFixedString           sound;            // 20 - size 08
+			BSTArray<BSFixedString> objectives;       // 28 - size 18
```
Comment (Ryan-rsm-McKenzie) [inline include/RE/H/HUDNotifications.h:23]:
```suggestion
			BSString                text;             // 00
			BSString                status;           // 10
			BSFixedString           sound;            // 20
			BSTArray<BSFixedString> objectives;       // 28
```

**Review by Ryan-rsm-McKenzie — CHANGES_REQUESTED** [review]

File: include/RE/H/HUDNotifications.h (hunk 5 above)
Comment (zax-ftw) [reply include/RE/H/HUDNotifications.h:28]: `pad44` initialized to 0 is mistake or intended?

Comment (Ryan-rsm-McKenzie) [reply include/RE/H/HUDNotifications.h:28]: It is an intentional decision to zero initialize all members of the struct.

File: include/RE/H/HUDNotifications.h [hunk 7]
```diff
@@ -45,11 +38,11 @@ namespace RE
...
 
 		// members
-		BSTArray<UnkData1> unk28;                  // 28
-		GFxValue           questNotificationRoot;  // 40 - "HUDMovieBaseInstance.QuestUpdateBaseInstance"
-		GFxValue           objectives;             // 58 - "QuestNotification.ObjectivesA"
-		std::uint32_t      unk70;                  // 70
-		float              levelPoints;            // 74
+		BSTArray<Notification> queue;        // 28 - size 18
+		GFxValue               root;         // 40 - size 18 - "HUDMovieBaseInstance.QuestUpdateBaseInstance"
+		GFxValue               objectives;   // 58 - size 18 - "QuestNotification.ObjectivesA"
```
Comment (zax-ftw) [inline include/RE/H/HUDNotifications.h:43]:
```suggestion
		BSTArray<Notification> queue;        // 28
		GFxValue               root;         // 40 - "HUDMovieBaseInstance.QuestUpdateBaseInstance"
		GFxValue               objectives;   // 58 - "QuestNotification.ObjectivesA"
```

File: include/RE/H/HUDNotifications.h [hunk 8]
```diff
@@ -12,29 +15,19 @@ namespace RE
 	public:
 		inline static constexpr auto RTTI = RTTI_HUDNotifications;
 
-		struct UnkData1
+		struct Notification
 		{
```
Comment (Ryan-rsm-McKenzie) [inline include/RE/H/HUDNotifications.h:19]:
```suggestion
		{
		public:
		// members
```

File: include/RE/H/HUDNotifications.h [hunk 9]
```diff
@@ -12,29 +15,19 @@ namespace RE
...
-			std::uint64_t      unk58;  // 58
+			BSString                text;              // 00
+			BSString                status;            // 10
+			BSFixedString           sound;             // 20
+			BSTArray<BSFixedString> objectives;        // 28
+			std::uint32_t           type{ 0 };         // 40
+			std::uint32_t           pad44{ 0 };        // 44
+			TESQuest*               quest{ nullptr };  // 48
+			TESWordOfPower*         word{ nullptr };   // 50
+			std::uint32_t           time{ 0 };         // 58 - gameTime + iObjectivesWaitTime
```
Comment (Ryan-rsm-McKenzie) [inline include/RE/H/HUDNotifications.h:28]:
```suggestion
			std::uint32_t           time{ 0 };         // 58 - gameTime + iObjectivesWaitTime
			std::uint32_t pad5C{ 0 }; // 5C
```

**Review by Ryan-rsm-McKenzie — CHANGES_REQUESTED** [review]

File: include/RE/H/HUDNotifications.h (hunk 8 above)
Comment (zax-ftw) [reply include/RE/H/HUDNotifications.h:19]: I changed it back to `struct` its public by default :)

Comment (Ryan-rsm-McKenzie) [reply include/RE/H/HUDNotifications.h:19]: I am aware that the default access specifier for `struct` is public. I would still like you to make this change for consistency across the code base.

File: CMakePresets.json (hunk 2 above)
Comment (zax-ftw) [reply CMakePresets.json]: May I suggest updating [ExamplePlugin](https://github.com/Ryan-rsm-McKenzie/ExamplePlugin-CommonLibSSE) with `VCPKG_INSTALLATION_ROOT` for more consistency?

File: include/RE/H/HUDNotifications.h [hunk 10]
```diff
@@ -1,9 +1,12 @@
...
 #include "RE/B/BSFixedString.h"
+#include "RE/B/BSString.h"
 #include "RE/B/BSTArray.h"
 #include "RE/G/GFxValue.h"
 #include "RE/H/HUDObject.h"
+#include "RE/T/TESQuest.h"
+#include "RE/T/TESWordOfPower.h"
 
 namespace RE
 {
```
Comment (Ryan-rsm-McKenzie) [inline include/RE/H/HUDNotifications.h:12]:
```suggestion
{

  class TESQuest;
  class TESWordOfPower;

```

File: include/RE/H/HUDNotifications.h [hunk 11]
```diff
@@ -1,9 +1,12 @@
 #pragma once
 
 #include "RE/B/BSFixedString.h"
+#include "RE/B/BSString.h"
 #include "RE/B/BSTArray.h"
 #include "RE/G/GFxValue.h"
 #include "RE/H/HUDObject.h"
+#include "RE/T/TESQuest.h"
+#include "RE/T/TESWordOfPower.h"
```
Comment (Ryan-rsm-McKenzie) [inline include/RE/H/HUDNotifications.h:9]:
```suggestion
```

**Review by Ryan-rsm-McKenzie — CHANGES_REQUESTED** [review]

**Review by Ryan-rsm-McKenzie — APPROVED** [review]

**Pull request merged**